server/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -152,8 +151,21 @@ public DocSchemaInfo(final String schemaName,
 
     @Override
     public TableInfo getTableInfo(String name) {
+        Metadata metadata = clusterService.state().metadata();
+        DocTableInfo docTableInfo = docTableByName.get(name);
         try {
-            return docTableByName.computeIfAbsent(name, n -> docTableInfoFactory.create(new RelationName(schemaName, n), clusterService.state().metadata()));
+            if (docTableInfo == null) {
+                return docTableByName.computeIfAbsent(
+                    name,
+                    n -> docTableInfoFactory.create(new RelationName(schemaName, n), metadata)
+                );
+            }
+            if (docTableInfo.metadataVersion() < metadata.version()) {
```

Comment:
Can we use IndexMetadata.getMappingVersion() here?  Or does that not work with the way we use index templates etc?  As I read it this means that any clusterstate update invalidates the doctableinfo for every table.

Reply:
The problem is that for empty partitioned tables there is no `IndexMetadata`. 
I can try to tune it to use the template version if it is partitioned, and otherwise the index metadata version